### ISSUE TITLE
C++20 concepts appear in namespace listing despite marked private

### ISSUE THREAD

**Issue by seijikun** [doxygen/doxygen#8981]
**Describe the bug**
The visibility of concepts seems to behave differently than for example with structs.
If I mark a struct with `/** @private */`, it's hidden from the documentation.
Though when I mark my concepts, they are shown in the `namespace listing` but clicking on them resolves to a dead link.
```c++
/**
 * @brief TestNamespace
 */
namespace TestNamespace {

	/** @private */
	template<typename T> concept is_pair = requires(T pair) {
		typename T::first_type;
		typename T::second_type;
		{std::get<0>(pair)} -> std::convertible_to<typename T::first_type>;
		{std::get<1>(pair)} -> std::convertible_to<typename T::second_type>;
	};

	/** @private */
	template<typename T> concept is_optional = requires(T optional, typename T::value_type value) {
		typename T::value_type;
		{optional.value()} -> std::convertible_to<typename T::value_type>;
		{optional.value_or(value)} -> std::convertible_to<typename T::value_type>;
		{optional = value};
	};
	
	/** @private */
	struct Undocumented {};
	
	/** @private */
	struct Undocumented2 {};

	/**
	 * @brief Documented
	 */
	template<typename TSelf> class Documented;
}
```

The concepts listing is properly hidden since both concepts are marked private, though if one concept is documented (and the private marker removed), the concept listing appears. Here, the private concept also appears with a dead link instead of being hidden.

**Expected behavior**
As long as `EXTRACT_PRIVATE` is set to `OFF`, I would have expected concepts to be hidden from listings, if marked `@private`.

**To Reproduce**
[concepts-visibility-bug.tar.gz](https://github.com/doxygen/doxygen/files/7771249/concepts-visibility-bug.tar.gz)

**Version**
`1.9.2` and `c94a1ee83db4eb098e72db3fe992ff4b8b9ace56` (really this time ;))

**Comment by albert-github** [comment]
You mean that in the file `concepts.html` of your attached example (that differs from the code in the question in the sense that in the question you have `is_pair` as `/** @private */` and in the attached example `is_pair` is documented as 
```
        /**
         * @brief DocumentedConcept
         */
```

you would not expect to see `is_optional` as shown in the image:

![image](https://user-images.githubusercontent.com/5223533/147340246-e3875f33-19d9-45bd-82ba-662c272a36ea.png)


It looks like that concepts don't take the `@private` into consideration as when when I document both `is_pair` and `is_optional` and set them post to `@private ` they still appear both and with proper links.
Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/7773761/example.tar.gz)

**Comment by seijikun** [comment]
Yes, sorry. The example I posted already applied the change mentioned in the comment:
> The concepts listing is properly hidden since both concepts are marked private, though if one concept is documented (and the private marker removed), the concept listing appears.
